src/ucp/core/ucp_am.c

```diff
@@ -969,6 +969,27 @@ ucp_am_params_check_memh(const ucp_request_param_t *param, uint32_t *flags_p)
     return UCS_OK;
 }
 
+static int
+ucp_am_send_nbx_check_header_length(ucp_worker_h worker, size_t header_length)
+{
+    ucp_worker_attr_t attr;
+
+    attr.field_mask = UCP_WORKER_ATTR_FIELD_MAX_AM_HEADER;
+    if (ucp_worker_query(worker, &attr) != UCS_OK) {
```

Comment:
IMO better use ucp_am_max_header_size. Or maybe even cache max header size on the worker?

Reply:
I think this is a better way in terms of code support. `ucp_am_send_nbx_check_header_length` is less sensitive to changes in other methods in this case.
A user obtains the value using `ucp_worker_query`. Which is an interface. `ucp_am_max_header_size` is an internal method. The implementation of `ucp_worker_query` may change, `ucp_am_max_header_size` may change its name, or may be gone. These are more likely events than the change of the interface.

Reply:
so, maybe cache then?

Reply:
ok, I will cache the value on the worker.

Reply:
Also moved `ucp_am_max_header_size` to worker.c, and renamed it to `ucp_worker_get_max_am_header`, because the method operates with a worker handler.

Reply:
IMO we should not call API functions from internal functions, it's violating SW layers

Reply:
using the field of UCP worker directly